### PR TITLE
Clang-tidy: google-readability-casting

### LINKED ISSUE #5744: clang-tidy google-readability-casting tule
https://clang.llvm.org/extra/clang-tidy/checks/google-readability-casting.html

> Finds usages of C-style casts.

I am in favor of using that clang-tidy rule. @maxGimeno Could you give it a try? First on a small set of CGAL, like a few CGAL plugins.

_Originally posted by @lrineau in https://github.com/CGAL/cgal/pull/5731#discussion_r642282476_

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

Apply google-readability-casting to the off plugin.
## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #5744

### CODE REVIEW

**Comment by lrineau** [comment]
The 3D demo, with all its OpenGL code, will be an heavy user of casts. Let's hope the patch will be smaller on CGAL headers.

**Comment by maxGimeno** [comment]
Appart from that, are you satisfied ? Should I run it on the whole demo ? A whole testsuite ?

**Comment by lrineau** [comment]
> Appart from that, are you satisfied ? Should I run it on the whole demo ? A whole testsuite ?

Yes: it seems to do what is expected. Have it run a few packages, like Mesh_2 and Mesh_3, to see how it deals with templated code. Then, if those tests are good, we will go on and continue on the whole CGAL testsuite.

**Comment by sloriot** [comment]
@maxGimeno could you please refresh this branch so that it can be tested? We will keep the scope of this PR to the demo. If you have time this week, you can open another for the rest of CGAL.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-42.shtml

File: .clang-tidy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-Checks:          '-clang-diagnostic*,-clang-analyzer*,modernize-use-nullptr'
```
Comment (lrineau) [inline .clang-tidy:2]: Why do we remove `modernize-use-nullptr`?!

Comment (sloriot) [reply .clang-tidy:2]: a mistake I'd say. I'll restore it (or you can do it) directly in master once I'm done with checks on another PR.

**Pull request merged**